code/modules/clothing/shoes/magboots.dm

```diff
@@ -126,3 +126,168 @@
 			..()
 		else
 			to_chat(user, "<span class='notice'>You poke the gem on [src]. Nothing happens.</span>")
+
+
+/obj/item/clothing/shoes/magboots/gravity
+	name = "gravitational boots"
+	desc = "These experimental boots try to get around the restrictions of magboots by installing miniture gravitational generators in the soles. Sadly, power hungry, and needs a gravitational anomaly core."
+	icon_state = "gravboots0"
+	origin_tech = "materials=6;magnets=6;engineering=6"
+	actions_types = list(/datum/action/item_action/toggle, /datum/action/item_action/gravity_jump) //In other news, combining magboots with jumpboots is a mess
+	strip_delay = 10 SECONDS
+	put_on_delay = 10 SECONDS
+	slowdown_active = SHOES_SLOWDOWN
+	magboot_state = "gravboots"
+	magpulse_name = "micro gravitational traction system"
+	var/datum/martial_art/grav_stomp/style = new //Only works with core and cell installed.
+	var/jumpdistance = 5
+	var/jumpspeed = 3
+	var/recharging_rate = 6 SECONDS
+	var/recharging_time = 0 // Time until next dash
+	var/dash_cost = 1000 // Cost to dash.
+	var/power_consumption_rate = 30 // How much power is used by the boots each cycle when magboots are active
+	var/obj/item/assembly/signaler/anomaly/grav/core = null
+	var/obj/item/stock_parts/cell/cell = null
+
+/obj/item/clothing/shoes/magboots/gravity/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/clothing/shoes/magboots/gravity/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	QDEL_NULL(style)
+	QDEL_NULL(cell)
+	QDEL_NULL(core)
+	return ..()
+
+/obj/item/clothing/shoes/magboots/gravity/examine(mob/user)
+	. = ..()
+	if(core && cell)
+		. += "<span class='notice'>[src] are fully operational!</span>"
+		. += "<span class='notice'>The boots are [round(cell.percent())]% charged.</span>"
+	else if(core)
+		. += "<span class='warning'>It has a gravitational anomaly core installed, but no power cell installed.</span>"
+	else if(cell)
+		. += "<span class='warning'>It has a power installed, but no gravitational anomaly core installed.</span>"
+	else
+		. += "<span class='warning'>It is missing a gravitational anomaly core and a power cell.</span>"
+
+/obj/item/clothing/shoes/magboots/gravity/attack_self(mob/user)
+	if(!cell)
+		to_chat(user, "<span class='warning'>Your boots do not have a power cell!</span>")
+		return
+	else if(cell.charge <= power_consumption_rate && !magpulse)
+		to_chat(user, "<span class='warning'>Your boots do not have enough charge!</span>")
+		return
+	if(!core)
+		to_chat(user, "<span class='warning'>There's no core installed!</span>")
+		return
+
```

Comment:
Instead of starting to process in init, I would have the processing only happen when the boots are on. So you can have this code here in attack_self
```dm
if(magpulse)
	STOP_PROCESSING(SSobj, src)
else
	START_PROCESSING(SSobj, src)
```
And then you can remove the `if(!cell) return` check, and the `&& magpulse` check in `process()` since we know it has a cell and magpulse is on.

Reply:
Can do